terraform-provider-constellation/internal/provider/provider.go

```diff
@@ -4,22 +4,26 @@ Copyright (c) Edgeless Systems GmbH
 SPDX-License-Identifier: AGPL-3.0-only
 */
 
+// The provider package implements the Constellation Terraform provider's
+// "provider" resource.
```

Comment:
Not really important for this PR, but this doc comment doesn't tell me anything about this package without already having knowledge about it.

Reply:
I agree. I think we should have a separate PR to make the provider nice for public.
Tracked in https://dev.azure.com/Edgeless/Edgeless/_backlogs/backlog/Edgeless%20Team/Features?workitem=3568